### PR TITLE
use hardcoded authorizer for metrics

### PR DESCRIPTION
**Pull request by deads2k**
This ensures we always get metrics even if something goes wrong requesting a SAR.  It also reduces query pressure.

### CODE REVIEW

**Comment by deads2k** [comment]
/hold 

fake bump

**Comment by deads2k** [comment]
/hold cancel

now a real bump

File: staging/src/k8s.io/apiserver/pkg/server/options/authorization.go [hunk 1]
```diff
@@ -161,6 +162,9 @@ func (s *DelegatingAuthorizationOptions) toAuthorizer(client kubernetes.Interfac
 		authorizers = append(authorizers, authorizerfactory.NewPrivilegedGroups(s.AlwaysAllowGroups...))
 	}
 
+	// add an authorizer to always approver the openshift metrics scraper.
+	authorizers = append(authorizers, hardcodedauthorizer.NewHardCodedMetricsAuthorizer())
```
Comment (p0lyn0mial) [inline staging/src/k8s.io/apiserver/pkg/server/options/authorization.go:166]: ah, okay this is for the `/metrics` endpoint

**Comment by p0lyn0mial** [comment]
@deads2k does it apply to all aggregated APIs ?

**Comment by p0lyn0mial** [comment]
@deads2k I don't see anything similar for the root API server. I think it is not needed since there is no RT and data needed for evaluation is in the cache anyway. Is that correct?

**Comment by deads2k** [comment]
> @deads2k I don't see anything similar for the root API server. I think it is not needed since there is no RT and data needed for evaluation is in the cache anyway. Is that correct?

Yes, it is correct.



> @deads2k does it apply to all aggregated APIs ?

They will need to adjust the authorizer independently.

**Comment by deads2k** [comment]
looks to be working

/hold cancel

**Comment by deads2k** [comment]
letting the carry come back for metrics availability.

**Pull request merged**